File: Projet-INF1013/src/app/models/person/person.ts
import {EmergencyContact} from '../emergency/emergency-contact';
import {Intervenant} from '../intervenant/intervenant';
import {Doctor} from '../doctor/doctor';
import {OtherPerson} from '../other-person/other-person';
import {User} from '../users/user';

export interface Person {
  interfaceName: 'Person';
  id: number;
  lname: string;
  fname: string;
  birthday: Date;
  sexe: 'Homme' | 'Femme';
  address: string;
  phone: number;
  NAS: number;
  healthIssues: string;
  workCityID: number; // Trois-Rivières, Shawinigan, Louiseville, St-Tite
  startDate: Date;
  endDate: Date;
  referenceID: number;
  residenceTypeID: number; // appartement, famille d’accueil, logement supervisé
  educationalLevelID: number; // aucun diplôme, études secondaires, études professionnelles, études collégiales, études universitaires
  programStartDate: Date;
  programEndDate: Date;
  departureReasonID: number;
  hoursPerDay: number;
  statusID: number;
  roamingTracking: boolean;
  roamingStartDate: Date;
  roamingEndDate: Date;
  communityWork: boolean;
  communityStartDate: Date;
  communityEndDate: Date;
  hourlyRate: number;
  transportFees: number;
  responsibleIntervenantID: number;
  emergencyContact: EmergencyContact;
  // followedBy: Intervenant | EmergencyContact | Doctor | OtherPerson;
  followedBy: User | EmergencyContact | Doctor | OtherPerson;
  active: boolean;
}

File: Projet_Angular_INF1013/backend/src/main/java/ca/uqtr/dmi/inf1013/model/Person.java
package ca.uqtr.dmi.inf1013.model;

import lombok.Data;

import javax.persistence.*;
import java.util.Date;

@Entity()
@Table(name = "Person")
@Data // Permet d'avoir les get set

public class Person {
  @Id() // Permet de dire que le champs qui suit est l'id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long ID;
  private String interfaceName;
  private String lname;
  private String fname;
  private Date birthday;
  private String sexe;
  private String address;
  private String phone;
  private Number nas;
  private String healthIssues;
  private Number workCityId;
  private Date startDate;
  private Date endDate;
  private Number referenceId;
  private Number residenceTypeId;
  private Number educationalLevelId;
  private Date programStartDate;
  private Date programEndDate;
  private Number departureReasonId;
  private Number hoursPerDay;
  private Number statusId;
  private Boolean roamingTracking;
  private Date roamingStartDate;
  private Date roamingEndDate;
  private Boolean communityWork;
  private Date communityStartDate;
  private Date communityEndDate;
  private Number hourlyRate;
  private Number transportFees;
  private Number responsibleIntervenantId;
  private Number emergencyContactId;
  private Number followedById;
  private Boolean active;
}
